tests/test_cli_compile.py

```diff
@@ -4044,3 +4044,63 @@ def test_second_order_requirements_relative_path_in_separate_dir(
             # via -r {output_path}
         """
     )
+
+
+@pytest.mark.parametrize("pyproject_path_is_absolute", (True, False))
+def test_that_self_referential_pyproject_toml_extra_can_be_compiled(
+    pip_conf, runner, tmp_path, monkeypatch, pyproject_path_is_absolute
+):
+    """
+    Test that a `pyproject.toml` source file can use self-referential extras
```

Comment:
Docstrings are usually parsed as RST:
```suggestion
    Test that a :file:`pyproject.toml` source file can use self-referential extras
```